packages/cli/src/ExternalSecrets/providers/azure-key-vault/azure-key-vault.ts

```diff
@@ -0,0 +1,129 @@
+import { ClientSecretCredential } from '@azure/identity';
+import { SecretClient } from '@azure/keyvault-secrets';
+import type { SecretsProvider, SecretsProviderState } from '@/Interfaces';
+import type { INodeProperties } from 'n8n-workflow';
+import type { AzureKeyVaultContext, AzureKeyVaultSecret } from './types';
+import { DOCS_HELP_NOTICE, EXTERNAL_SECRETS_NAME_REGEX } from '@/ExternalSecrets/constants';
+
+export class AzureKeyVault implements SecretsProvider {
+	name = 'azureKeyVault';
+
+	displayName = 'Azure Key Vault';
+
+	state: SecretsProviderState = 'initializing';
+
+	properties: INodeProperties[] = [
+		DOCS_HELP_NOTICE,
+		{
+			displayName: 'Vault Name',
+			description: 'The name of your existing Azure Key Vault.',
+			name: 'vaultName',
+			type: 'string',
+			default: '',
+			required: true,
+			placeholder: 'e.g. my-vault',
+			noDataExpression: true,
+		},
+		{
+			displayName: 'Directory (Tenant) ID',
+			name: 'tenantId',
+			type: 'string',
+			default: '',
+			required: true,
+			placeholder: 'e.g. 7dec9324-7074-72b7-a3ca-a9bb3012f466',
+			noDataExpression: true,
+		},
+		{
+			displayName: 'Application (Client) ID',
+			name: 'clientId',
+			type: 'string',
+			default: '',
+			required: true,
+			placeholder: 'e.g. 7753d8c2-e41f-22ed-3dd7-c9e96463622c',
+			typeOptions: { password: true },
+			noDataExpression: true,
+		},
+		{
+			displayName: 'Client Secret',
+			name: 'clientSecret',
+			description: 'The client secret value of your registered application.',
+			type: 'string',
+			default: '',
+			required: true,
+			typeOptions: { password: true },
+			noDataExpression: true,
+		},
```

Comment:
The `displayName` is being ignore for these because they are hard coded for `clientId` and `clientSecret` in the FE:

https://github.com/n8n-io/n8n/blob/f4dcc0751ecc3f3696bf4ba98e7377ba593ffcba/packages/editor-ui/src/plugins/i18n/index.ts#L120-L125

Reply:
To be pragmatic we could use the `hint` to show the information:

![image](https://github.com/user-attachments/assets/68d638b9-9fcb-4bc4-bc2b-3d907e2b52eb)

```ts
	properties: INodeProperties[] = [
		DOCS_HELP_NOTICE,
		{
			displayName: 'Vault Name',
			hint: 'The name of your existing Azure Key Vault.',
			name: 'vaultName',
			type: 'string',
			default: '',
			required: true,
			placeholder: 'e.g. my-vault',
			noDataExpression: true,
		},
		{
			displayName: 'Directory (Tenant) ID',
			name: 'tenantId',
			type: 'string',
			default: '',
			required: true,
			placeholder: 'e.g. 7dec9324-7074-72b7-a3ca-a9bb3012f466',
			noDataExpression: true,
		},
		{
			displayName: 'Application (Client) ID',
			name: 'clientId',
			hint: 'In azure this can be called "Application (Client) ID".',
			type: 'string',
			default: '',
			required: true,
			placeholder: 'e.g. 7753d8c2-e41f-22ed-3dd7-c9e96463622c',
			typeOptions: { password: true },
			noDataExpression: true,
		},
		{
			displayName: 'Client Secret',
			name: 'clientSecret',
			hint: 'The client secret value of your registered application.',
			type: 'string',
			default: '',
			required: true,
			typeOptions: { password: true },
			noDataExpression: true,
		},
	];
```